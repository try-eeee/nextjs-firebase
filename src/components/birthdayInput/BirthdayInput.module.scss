.birthdayInput {
  width: 100%;
  padding: 0.75rem;
  border: 1px solid #ccc;
  border-radius: 4px;
  margin-bottom: 1rem;
  font-size: 1rem;
}
