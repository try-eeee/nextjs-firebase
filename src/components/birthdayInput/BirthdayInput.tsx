import styles from "./BirthdayInput.module.scss";
import { UseFormRegisterReturn } from "react-hook-form";

interface BirthdayInputProps extends React.ComponentPropsWithoutRef<"input"> {
  register?: UseFormRegisterReturn;
}

export function BirthdayInput(props: BirthdayInputProps) {
  const { register, ...rest } = props;
  return <input {...register} {...rest} className={styles.birthdayInput} />;
}
