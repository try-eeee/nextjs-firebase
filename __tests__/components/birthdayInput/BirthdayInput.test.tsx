import React from "react";
import { render } from "@testing-library/react";
import { BirthdayInput } from "@/components/birthdayInput/BirthdayInput";

describe("BirthdayInput UnitTest", () => {

  test("Snapshot Test Of BirthdayInput", () => {
    const { asFragment } = render(<BirthdayInput />);
    expect(asFragment()).toMatchSnapshot();
  });
});
