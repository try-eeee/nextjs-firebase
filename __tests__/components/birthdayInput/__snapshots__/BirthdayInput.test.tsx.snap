// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BirthdayInput UnitTest Snapshot Test Of BirthdayInput 1`] = `
<DocumentFragment>
  <input
    class="birthdayInput"
  />
</DocumentFragment>
`;
